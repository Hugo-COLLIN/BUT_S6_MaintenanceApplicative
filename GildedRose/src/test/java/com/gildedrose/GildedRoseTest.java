package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void testUpdateQuality() {
//        Approvals.verify(doUpdateQuality("foo", 0, 0));
         CombinationApprovals.verifyAllCombinations(
                this::doUpdateQuality,
                new String[] { "foo", "Aged Brie" },
                new Integer[] { 0, 5 },
                new Integer[] { 0 });
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }

}
