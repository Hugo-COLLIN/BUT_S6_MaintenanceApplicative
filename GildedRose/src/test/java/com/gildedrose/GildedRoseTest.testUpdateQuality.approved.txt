[foo, 0, 0] => foo, -1, 0 
[foo, 5, 0] => foo, 4, 0 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
